Add compatibility notes for using ActionText with ViewComponent (#990)
github/view_component: docs/compatibility.md

## Bridgetown (Static Site Generator)

[Bridgetown](https://www.bridgetownrb.com/) supports ViewComponent via an experimental shim provided by the [bridgetown-view-component gem](https://github.com/bridgetownrb/bridgetown-view-component). More information available [here](https://www.bridgetownrb.com/docs/components/ruby#need-compatibility-with-rails-try-viewcomponent-experimental).

> ## ActionText Support

Reviewer: ```suggestion
## ActionText
```

> Trying to use `rich_text_area` via ActionText out of the box with ViewComponent will result in this error:

Reviewer: ```suggestion
Using `rich_text_area` from ActionText in a ViewComponent will result in this error:
```

> ```
> undefined method `rich_text_area_tag'
> ```

Reviewer: ```suggestion

```
undefined method `rich_text_area_tag'
```
```

Reviewer 2: Hey @joelhawksley, I don't think this suggestion applied correctly due to using ``` inside the suggestion block

> This is because ViewComponent does not have access to the helpers it needs via ActionText. To avoid this, add the following to your component (or base component):

Reviewer: ```suggestion
This is due to ViewComponent not having access to the helpers it needs via ActionText. As a workaround, add the following to your component (or base component):
```

## Bridgetown (Static Site Generator)

[Bridgetown](https://www.bridgetownrb.com/) supports ViewComponent via an experimental shim provided by the [bridgetown-view-component gem](https://github.com/bridgetownrb/bridgetown-view-component). More information available [here](https://www.bridgetownrb.com/docs/components/ruby#need-compatibility-with-rails-try-viewcomponent-experimental).

## ActionText Support

Trying to use `rich_text_area` via ActionText out of the box with ViewComponent will result in this error:
```
undefined method `rich_text_area_tag'
```

This is because ViewComponent does not have access to the helpers it needs via ActionText. To avoid this, add the following to your component (or base component):

```
delegate :rich_text_area_tag, to: :helpers
```

> *_This is the simplest route. Adding the helpers directly via `include ActionText::Engine.helpers` leads to several more errors and requires additional configuration that is not necessary._

Reviewer 3: Looks great but I'm wondering if it's worth mentioning the other method without saying what it is or linking to it? Imo I believe this to be the "right" way to fix this and it doesn't just pertain to ActionText!

[Here's a great example](https://www.bridgetownrb.com/docs/components/ruby#helpers-1)

Author: Good call! I went ahead and removed it.
